src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -464,7 +464,7 @@ def _to_data_or_iter(
     ) -> Dict[str, Any]:
         qid = results_cursor.sfqid
         if (
-            to_iter and not to_pandas
+            to_iter
         ):  # Fix for SNOW-869536, to_pandas doesn't have this issue, SnowflakeCursor.fetch_pandas_batches already handles the isolation.
             new_cursor = results_cursor.connection.cursor()
```

Comment:
let's remove this comment as well since it turns out `to_pandas` also suffers from the same problem